crates/llama-cpp-bindings/src/llama.rs

```diff
@@ -126,13 +109,25 @@ fn start_llama_service_impl(
 
 pub struct LlamaService {
     tx: Sender<LlamaInitRequest>,
+    next_request_id: RwLock<u32>,
 }
 
 impl LlamaService {
     pub fn new(engine: UniquePtr<ffi::TextInferenceEngine>) -> Self {
         let (tx, rx) = channel(1);
         start_llama_service_impl(engine, rx);
-        Self { tx }
+        Self {
+            tx,
+            next_request_id: RwLock::default(),
+        }
+    }
+
+    async fn alloc_request_id(&self) -> u32 {
+        let ret = *self.next_request_id.read().await;
+
+        // 1048576 (2^20) should be large enough to avoid any collision.
+        *self.next_request_id.write().await = (ret + 1) % 1048576;
```

Comment:
The comment doesn't explain the purpose of this code clearly enough, why are you adding 1 and performing modulus?

Reply:
request_id represents an unique request. It shouldn't being conflicting with requests that being concurrently processed.

In tabby's case, since it's unlikely it can handle 2^20 requests in parallel, thus 2^20 modulus should be good enough to generate unique request ids.

Reply:
```suggestion
        // request_id must be unique at time of processing and shouldn't overflow, tabby is unlikely to handle 2^20 requests at once
        *self.next_request_id.write().await = (ret + 1) % 1048576;
```